### PR TITLE
fix: properly check kubectl success on init. Force workspace folder deletion

### PR DESCRIPTION
**Pull request by 89luca89**
Right now the k8s init wasn't working properly, allowing for invalid kubeconfig paths, and values
Contextually this would lead to `delete` to not actually delete the workspace dir, thus making the user stuck with the "wrong" config, not able to update it and requiring manual intervention.

Resolves ENG-1600

### CODE REVIEW

File: cmd/delete.go [hunk 1]
```diff
@@ -112,7 +112,10 @@ func (cmd *DeleteCmd) Run(ctx context.Context, devPodConfig *config.Config, args
 		GracePeriod: duration,
 	})
 	if err != nil {
-		return err
+		err2 := clientimplementation.DeleteWorkspaceFolder(client.Context(), client.Workspace(), log.Default)
+		if err2 != nil {
+			return errors.Wrap(err2, err.Error())
+		}
```
Comment (ThomasK33) [inline cmd/delete.go:118]: How about:

```suggestion
		return errors.Join(err, clientimplementation.DeleteWorkspaceFolder(client.Context(), client.Workspace(), log.Default))
```

At the end of the day we've encountered an error and should probably return, independent of the workspace delete failing or not.

Comment (89luca89) [reply cmd/delete.go]: Done

**Review by ThomasK33 — APPROVED** [review]
LGTM

File: cmd/delete.go [hunk 2]
```diff
@@ -112,7 +112,7 @@ func (cmd *DeleteCmd) Run(ctx context.Context, devPodConfig *config.Config, args
 		GracePeriod: duration,
 	})
 	if err != nil {
-		return err
+		return errors.Wrap(err, clientimplementation.DeleteWorkspaceFolder(client.Context(), client.Workspace(), log.Default).Error())
```
Comment (FabianKramm) [inline cmd/delete.go:115]: I'm not sure this is a good idea, we have the `--force` flag to make sure even if the deletion fails the workspace is deleted.

**Review by FabianKramm — CHANGES_REQUESTED** [review]

File: cmd/delete.go (hunk 2 above)
Comment (89luca89) [reply cmd/delete.go:115]: This was not working with the `--force` flag, even with it, it was stopping at this very point

File: providers/kubernetes/provider.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 name: kubernetes
-version: v0.0.1
+version: v0.0.2
```
[the remark this answers is not in the document]
Comment (89luca89) [reply providers/kubernetes/provider.yaml:2]: ah ok I thought it was needed for the update

File: providers/kubernetes/provider.yaml [hunk 4]
```diff
@@ -95,11 +95,11 @@ agent:
 exec:
   init: |-
     # Check if kubernetes is reachable
-    KUBERNETES_REACHABLE=$(KUBECONFIG=${KUBERNETES_CONFIG} ${KUBECTL_PATH} get pods --namespace=${KUBERNETES_NAMESPACE} --context=${KUBERNETES_CONTEXT} > /dev/null 2>&1 && echo -n "true" || echo -n "false")
-    if [ "KUBERNETES_REACHABLE" = "false" ]; then
+    KUBERNETES_REACHABLE=$(${KUBECTL_PATH} get pods --kubeconfig=${KUBERNETES_CONFIG} --namespace=${KUBERNETES_NAMESPACE} --context=${KUBERNETES_CONTEXT} > /dev/null 2>&1 && echo -n "true" || echo -n "false")
```
[the remark this answers is not in the document]
Comment (89luca89) [reply providers/kubernetes/provider.yaml:98]: Using `--kubeconfig=` is the method we use inside the provider commands, and in the driver
This for example makes `~/.kube/config` behave differently between `init` and the other commands, as here, the `KUBECONFIG=${KUBERNETES_CONFIG}` will expand the `~` while `--kubeconfig=` won't, and will fail afterwards

This unifies the behaviour between the init and the rest of the commands

**Review by FabianKramm — APPROVED** [review]

**Pull request merged**